src/all/pixiv/src/eu/kanade/tachiyomi/extension/all/pixiv/Pixiv.kt

```diff
@@ -0,0 +1,212 @@
+package eu.kanade.tachiyomi.extension.all.pixiv
+
+import android.util.LruCache
+import eu.kanade.tachiyomi.network.asObservable
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.model.UpdateStrategy
+import eu.kanade.tachiyomi.source.online.HttpSource
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.Jsoup
+import rx.Observable
+import uy.kohesive.injekt.injectLazy
+import java.net.URLEncoder
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+private val json: Json by injectLazy()
+private val dateFormat by lazy { SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss", Locale.JAPAN) }
+
+private fun illustUrlToId(url: String): String =
+    url.substringAfterLast("/")
+
+private fun parseTimestamp(string: String) =
+    runCatching { dateFormat.parse(string)?.time!! }.getOrDefault(0)
+
+class Pixiv(override val lang: String) : HttpSource() {
+    override val name = "Pixiv"
+    override val baseUrl = "https://www.pixiv.net"
+    override val supportsLatest = true
+
+    private val siteLang: String = if (lang == "all") "ja" else lang
+    private val illustCache = LruCache<String, PixivIllust>(50)
+
+    override fun headersBuilder() = super.headersBuilder()
+        .add("Referer", "$baseUrl/")
+
+    class SortFilter : Filter.Sort("Sort", arrayOf("Date Uploaded"), Selection(0, false))
+    class NonMangaFilter : Filter.CheckBox("Include non-manga posts", false)
+
+    private fun ajax(method: String, path: String, params: Map<String, String> = emptyMap()): Request {
```

Comment:
Technically, the extension doesn't do any AJAX requests, so better to rename this method.

```suggestion
    private fun apiRequest(method: String, path: String, params: Map<String, String> = emptyMap()): Request {
```